pyinfra/modules/virtualenv.py

```diff
@@ -0,0 +1,44 @@
+# pyinfra
+# File: pyinfra/modules/pip.py
+# Desc: manage virtualenvs
+
+'''
+Manage Python virtual environments
+'''
+
+from __future__ import unicode_literals
+
+from pyinfra.api import operation
+from pyinfra.modules import files
+
+
+@operation
+def virtualenv(
+    state, host,
+    path, python='python3', site_packages=False, always_copy=False,
+    present=True,
+):
+    '''
+    Manage virtualenv.
+
+    + python: python interpreter to use
+    + site_packages: give access to the global site-packages
+    + always_copy: always copy files rather than symlinking
+    + present: whether the virtualenv should be installed
+    '''
+
+    if present is False and host.fact.directory(path):
+        # Ensure deletion of unwanted virtualenv
+        # no 'yield from' in python 2.7
+        for cmd in files.directory(state, host, path, present=False):
+            yield cmd
+
+    elif present and not host.fact.directory(path):
+        # Create missing virtualenv
+        command = '/usr/bin/virtualenv -p {}'.format(python)
+        if site_packages:
+            command += ' --system-site-packages'
+        if always_copy:
+            command += ' --always-copy'
+        command += ' ' + path
```

Comment:
Can these commands be changed into a list and use join (just to be consistent with the other modules); and also not explicitly list the location of the `virtualenv` bin as it may be installed elsewhere on some systems, eg:

```py
args = ['virtualenv']

if python:
    args.append('-p {0}'.format(python))
...

command = ' '.join(args)
```